lib/pyfrc/mains/cli_create_physics.py

```diff
@@ -0,0 +1,140 @@
+#
+# The idea here is you provide a simulation object that overrides specific
+# pieces of WPILib, and modifies motors/sensors accordingly depending on the
+# state of the simulation. An example of this would be measuring a motor
+# moving for a set period of time, and then changing a limit switch to turn
+# on after that period of time. This can help you do more complex simulations
+# of your robot code without too much extra effort.
+#
+# Examples can be found at https://github.com/robotpy/examples
+
+
+class PhysicsEngine:
+    """
+       Simulates a 4-wheel robot using Tank Drive joystick control
+    """
+
+    def __init__(self, physics_controller):
+        """
+            :param physics_controller: `pyfrc.physics.core.Physics` object
+                                       to communicate simulation effects to
+        """
+
+        self.physics_controller = physics_controller
+
+        """
+        # Change these parameters to fit your robot!
+        bumper_width = 3.25 * units.inch
+
+        self.drivetrain = tankmodel.TankModel.theory(
+            motor_cfgs.MOTOR_CFG_CIM,           # motor configuration
+            110 * units.lbs,                    # robot mass
+            10.71,                              # drivetrain gear ratio ωIn/ωOut
+            2,                                  # motors per side„
+            22 * units.inch,                    # robot wheelbase
+            23 * units.inch + bumper_width * 2, # robot width
+            32 * units.inch + bumper_width * 2, # robot length
+            6 * units.inch,                     # wheel diameter
+        )
+        """
+
+    def update_sim(self, hal_data, now, tm_diff):
+        """
+            Called when the simulation parameters for the program need to be
+            updated.
+            
+            :param now: The current time as a float
+            :param tm_diff: The amount of time that has passed since the last
+                            time that this function was called
+        """
+
+        # Simulate the drivetrain
+        """
+        lr_motor = hal_data["pwm"][1]["value"]
+        rr_motor = hal_data["pwm"][2]["value"]
+
+        # Not needed because front and rear should be in sync
+        # lf_motor = hal_data['pwm'][3]['value']
+        # rf_motor = hal_data['pwm'][4]['value']
+
+        x, y, angle = self.drivetrain.get_distance(lr_motor, rr_motor, tm_diff)
+        self.physics_controller.distance_drive(x, y, angle)
+        """
+'''
+
+sim_starter = """{
+	"pyfrc": {
+		"robot": {
+			"w": 3,
+			"h": 2,
+			"starting_x": 10,
+			"starting_y": 10,
+			"starting_angle": 0
+		},
+		"pwm": {
+			"1": "Left Rear",
+			"2": "Right Rear",
+			"3": "Left Front",
+			"4": "Right Front"
+		},
+		"joysticks": {
+			"0": {
+				"axes": {
+					"Y": "Left Side"
+				},
+				"buttons": {}
+			},
+			"1": {
+				"axes": {
+					"Y": "Right Side"
+				},
+				"buttons": {}
+			}
+		}
+	}
+}
+"""
```

Comment:
I think it'll be better to have the config here be a dict and serialise it when we run. Makes it a lot easier to then be able to ask the user questions about their robot.

Probably better to not assume a 2 joystick tank drive unless we ask the user either. My team has had drivers that prefer a gamepad...

Reply:
Are those tab characters? ... yeah, I agree with @auscompgeek , use a dict + json.dump

Reply:
@virtuald yeah I copied from one of the sample jsons. 

Ok